RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc

```diff
@@ -129,25 +161,59 @@ bool BeamSpotOnlineProducer::shouldShout(const edm::Event& iEvent) const {
   return true;  // Default to "shout" if the record is missing
 }
 
-bool BeamSpotOnlineProducer::processTransientRecord(const edm::EventSetup& iSetup,
-                                                    reco::BeamSpot& result,
-                                                    bool shoutMODE) {
-  auto const& spotDB = iSetup.getData(beamTransientToken_);
+bool BeamSpotOnlineProducer::processRecords(const edm::Event& iEvent,
+                                            const edm::EventSetup& iSetup,
+                                            reco::BeamSpot& result,
+                                            bool shoutMODE) {
+  auto const spotDBLegacy = getBeamSpotFromRecord(beamTokenLegacy_, iEvent, iSetup);
+  auto const spotDBHLT = getBeamSpotFromRecord(beamTokenHLT_, iEvent, iSetup);
+  auto const& spotDB = chooseBS(spotDBLegacy, spotDBHLT);
 
-  if (spotDB.beamType() != 2) {
-    if (shoutMODE && beamTransientRcdESWatcher_.check(iSetup)) {
+  if (spotDB.beamType() != reco::BeamSpot::Tracker) {
+    if (shoutMODE && (beamLegacyRcdESWatcher_.check(iSetup) || beamHLTRcdESWatcher_.check(iSetup))) {
       edm::LogWarning("BeamSpotOnlineProducer")
           << "Online Beam Spot producer falls back to DB value due to fake beamspot in transient record.";
     }
     return true;  // Trigger fallback to DB
   }
 
   // Create BeamSpot from transient record
-  createBeamSpotFromTransient(spotDB, result);
+  createBeamSpotFromRecord(spotDB, result);
   return false;  // No fallback needed
 }
 
-void BeamSpotOnlineProducer::createBeamSpotFromTransient(const BeamSpotObjects& spotDB, reco::BeamSpot& result) const {
+template <typename TokenType>
+const BeamSpotOnlineObjects BeamSpotOnlineProducer::getBeamSpotFromRecord(const TokenType& token,
```

Comment:
```suggestion
const BeamSpotOnlineObjects& BeamSpotOnlineProducer::getBeamSpotFromRecord(const TokenType& token,
```
?

Reply:
I will add this in the next commit. It also requires modifications in the function body

Reply:
done